agent/runner/jobs/mongodb_backup_job.go

```diff
@@ -94,6 +98,10 @@ func (j *MongoDBBackupJob) Run(ctx context.Context, send Send) error {
 		return errors.Wrapf(err, "lookpath: %s", pbmBin)
 	}
 
+	if j.dataModel != backupv1beta1.DataModel_PHYSICAL && j.dataModel != backupv1beta1.DataModel_LOGICAL {
+		return errors.Errorf("'%s' is not a supported data model for backups", j.dataModel)
+	}
```

Comment:
I believe constructor is a better place for that check. Let's move it to `NewMongoDBBackupJob`.